plugins/ltac2/tac2core.ml

```diff
@@ -1869,7 +1869,7 @@ let () = add_expr_scope "clause" q_clause Tac2quote.of_clause
 let () = add_expr_scope "hintdb" q_hintdb Tac2quote.of_hintdb
 let () = add_expr_scope "occurrences" q_occurrences Tac2quote.of_occurrences
 let () = add_expr_scope "dispatch" q_dispatch Tac2quote.of_dispatch
-let () = add_expr_scope "strategy" q_strategy_flag Tac2quote.of_strategy_flag
+let () = add_expr_scope "reductions" q_reductions Tac2quote.of_strategy_flag
```

Comment:
This is a user-facing change, unrelated to the internal name of the entry. You have to keep the old scope name.

Reply:
It will be confusing for users to have to mentally map "reductions" in the doc to "strategy" in the `Ltac2 Notation`.  I should be able to create a deprecation message to handle this (and keep `q_strategy_flag` around for another release).  WDYT?

Reply:
I think user-facing changes like this should, at the very least, be factored into a separate PR.  (Note that there's also the Ltac1 Notation strategy scope, IIRC). And for Fiat Crypto backwards compatibility it might be important to keep the `strategy` alias around for more than a single release (though I suppose we could include a version-dependent compat file)

Reply:
@jfehrle I am formally opposed to coupling the internal name of grammar entries with the one of Ltac2 scopes.

Reply:
> I think user-facing changes like this should, at the very least, be factored into a separate PR.

Why?

> (Note that there's also the Ltac1 Notation strategy scope, IIRC).

Meaning `Tactic Notation`.

> And for Fiat Crypto backwards compatibility it might be important to keep the `strategy` alias around for more than a single release (though I suppose we could include a version-dependent compat file)

If the alias is eventually removed, you would have to do a version-specific file so your old versions would still work.  AFAIK Coq's compatibility strategy has been to support forward compatibility rather than guaranteeing that older versions of packages will work on newer versions of Coq.  The latter is far more restrictive.

Reply:
This discussion does not make sense because Ltac2 scopes and grammar entries are not in one-to-one correspondance. There is more to programming languages than parsing.

Reply:
> I am formally opposed to coupling the internal name of grammar entries with the one of Ltac2 scopes.

Perhaps you would explain why you oppose this?  Not doing this leaves a small burden on Ltac2 users as I explained.

Reply:
> > I think user-facing changes like this should, at the very least, be factored into a separate PR.
> 
> Why?

Because I think @Zimmi48 and I both support the changes that do not impact .v files, and as I understand it @ppedrot will not block such changes, so I expect this part of the PR (which I think is the bulk of it, right?) can be merged soon.

The user-facing changes (the ones that show up in .v files) are more contentious: I'm not sold on them (though not strongly opposed) and @ppedrot seems strongly opposed, so I expect that part of the PR will need lots more discussion?

Reply:
> This discussion does not make sense because Ltac2 scopes and grammar entries are not in one-to-one correspondence.

I'm not suggesting that as a general rule nor any enforcement mechanism.  For `Tactic Notation`, for example, `constr` and `uconstr` have identical syntax but different semantics, so they have to have different names.  When that's not the case, having two names for the same syntax+semantics doesn't help anyone.

Reply:
It's part of a longer term plan to have a small DSL of scopes that allows more abstraction than what we have today. In this model, scopes have qualified names which makes zero sense w.r.t. the internal string name of an OCaml entry. I really don't understand why you insist so much on adding strong coupling on things that do not even live at the same level nor even in the same language, for that matter. Please stop pushing for design choices you don't seem to understand the consequences of. I have already lost too much time with the Ltac debugger debacle and the underlying problem hasn't even been solved yet.

Reply:
How would I know about your longer term plan if you don't mention it?  Or the consequences?  If you plan to change the names with the introduction of a DSL, that's a reasonable reason not to do this.

> I really don't understand why you insist so much on adding strong coupling on things that do not even live at the same level nor even in the same language, for that matter.

I only insist on having some explanation beyond "don't do that".  My point is that users see things across environments.  Consistency helps them.